dapp_runner/runner/runner.py

```diff
@@ -77,6 +78,9 @@ def __init__(self, config: Config, dapp: DappDescriptor):
         self.state_queue = asyncio.Queue()
         self.command_queue = asyncio.Queue()
         self._startup_finished = False
+        self._desired_app_state = ServiceState.pending
```

Comment:
given that the enumerated states of the app as a whole don't necessarily map to our service states, I wouldn't use the same state machine to describe it but rather use a separate set of states even if those sets were to start as identical

though, again, e.g. I'm unsure if the app state is ever e.g. "pending", which brings me to the other point -> the desired app state here should, I believe, be `running` already 

in the context of a single service - "pending" indicates that an instance of a service has already been requested and its status should be tracked but there is yet no agreement nor activity bound to it ... 

as we have no concept of an agreement, activity or any other way of having more concrete handles with regards to the app as a whole, my understanding is that there's no need or benefit from differentiation of pending vs starting in case of the whole app and any app should be "starting" immediately after the runner has started its job...